modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/DefaultExecutionMediator.java

```diff
@@ -90,29 +103,54 @@ public Object execute(OperationExecutor executor, OperationContext context) thro
         }
     }
 
-    private MutableConfigurationStats getMutableConfigurationStats(OperationContext context)
+    private Object executeWithRetryPolicy(OperationExecutor executor, OperationContext context, List<Interceptor> interceptors) throws Throwable
     {
-        ConfigurationStats stats = context.getConfiguration().getStatistics();
-        return stats instanceof MutableConfigurationStats
-               ? (MutableConfigurationStats) stats
-               : null;
+        final RetryPolicyTemplate retryPolicyTemplate;
+        final Optional<ConnectionProvider> connectionProvider;
+
+        connectionProvider = context.getConfiguration().getConnectionProvider();
+
+        retryPolicyTemplate = getRetryPolicyTemplateFromConnectionProvider(connectionProvider);
+        LifecycleUtils.initialiseIfNeeded(retryPolicyTemplate, muleContext);
```

Comment:
i know you're doing this to inject the mule context, but be careful! this also invokes the initialise() method. You shouldn't do that constantly unless you're certain that new instances are always used.